### PR TITLE
Newsletter

### PR DESCRIPTION
**Pull request by gabrielsaruhashi**
**Newsletter**
image
image
image

image

**Contacts**

image
image
image

### CODE REVIEW

File: src/components/header/Navbar.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { Navbar, Nav } from 'react-bootstrap';
 import { Link } from 'react-router-dom';
 import './Navbar.css';
-import { ReactComponent as Logo } from '../../assets/logo.svg';
+import { ReactComponent as Logo } from 'src/assets/logo.svg';
```
Comment (mpekala23) [inline src/components/header/Navbar.tsx:5]: 😍 amazing

File: src/components/pages/Contacts/index.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) =>
+  bindActionCreators(
+    { loadOrgContacts, addFilter, removeFilter, loadTags },
+    dispatch,
+  );
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+type PropsFromRedux = ConnectedProps<typeof connector>;
```
Comment (mpekala23) [inline src/components/pages/Contacts/index.tsx:34]: ohhhh this is cool i like this

File: src/components/pages/Contacts/index.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [filteredOrgContact, setFilteredOrgContacts] = useState<OrgContact[]>(
+    [],
+  );
+  const [hasFetchedData, setHasFetchedData] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!hasFetchedData && orgContacts.length === 0) {
+      loadOrgContacts();
+      loadTags();
```
Comment (mpekala23) [inline src/components/pages/Contacts/index.tsx:54]: do you want to `await` either of these calls?

File: src/components/pages/NewsletterHistory/index.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+}) => {
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [hasFetched, setHasFetched] = useState<boolean>(false);
+  const [filteredNewsletters, setFilteredNewsletters] = useState<
+    NewsletterLog[]
+  >([]);
+
+  useEffect(() => {
+    if (!hasFetched && newsletters.length === 0) {
+      loadNewsletters();
```
Comment (mpekala23) [inline src/components/pages/NewsletterHistory/index.tsx:36]: do you want an `await` here?

File: src/components/pages/UploadContacts/index.tsx [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+const mapStateToProps = (state: RootState) => ({
+  uploadedCsv: state.orgContacts.uploadedCsv,
+  uploadStep: state.orgContacts.uploadStep,
+  tags: state.tags.tags,
+  selectedTags: state.orgContacts.uploadSelectedTags,
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) =>
+  bindActionCreators(
```
Comment (mpekala23) [inline src/components/pages/UploadContacts/index.tsx:37]: ohhhh fancy, what exactly does `bindActionCreators` do? (looks cool)

File: src/components/pages/UploadContacts/index.tsx [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+          {uploadStep === 3 && (
+            <div className="upload-file-step-container">
+              <span className="black-500 p3 mb-3 font-weight-semibold">
+                We're adding your contacts!
+              </span>
+              <Clock size={60} color="#C3DBF5" />
+              <span className="mt-3 text-center">
+                Depending on the number of contacts you've uploaded, this may
+                take several minutes. You can move on to your next tasks in the
+                mean time.
```
Comment (mpekala23) [inline src/components/pages/UploadContacts/index.tsx:231]: i'm kinda too lazy to dig into this, but how would the user be notified if they moved on and then the uploaded succeeded / failed? Will an alert pop up on screen?

File: src/components/tags/Tag.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { generateTagColor } from '../../utils/utils';
+import { sampleTagColors } from '../../data/TagColors';
+import { X } from 'react-feather';
+
+interface TagProps {
+  label: string;
+  count?: number;
+  canRemove?: boolean;
+  // showCount: boolean;
```
Comment (mpekala23) [inline src/components/tags/Tag.tsx:10]: can this be removed? or maybe add a TODO?

File: src/components/pages/UploadContacts/index.tsx (hunk 5 above)
Comment (gabrielsaruhashi) [reply src/components/pages/UploadContacts/index.tsx:37]: Oh hmm, I saw it on a guide and it seemed like a good practise, but maybe it's not the case?
----
Turns an object whose values are action creators, into an object with the same keys, but with every action creator wrapped into a dispatch call so they may be invoked directly.

Normally you should just call dispatch directly on your Store instance. If you use Redux with React, react-redux will provide you with the dispatch function so you can call it directly, too
--- 
How would you do it?

File: src/components/pages/UploadContacts/index.tsx (hunk 6 above)
Comment (gabrielsaruhashi) [reply src/components/pages/UploadContacts/index.tsx:231]: Don't have that functionality yet! but my plan is to add some variables to the redux store to let me know once a big request like this is done. I'll add it as a todo

File: src/components/pages/Contacts/index.tsx (hunk 3 above)
Comment (gabrielsaruhashi) [reply src/components/pages/Contacts/index.tsx:54]: synced offline! looks like await is not needed here

File: src/components/pages/NewsletterHistory/index.tsx (hunk 4 above)
Comment (gabrielsaruhashi) [reply src/components/pages/NewsletterHistory/index.tsx:36]: same as before!

**Pull request merged**